src/Form/index.tsx

```diff
@@ -1920,11 +1918,21 @@ function Form({
       } else if (type === ACTION_LOGOUT) await Auth.inferAuthLogout();
       else if (type === ACTION_NEW_SUBMISSION) await updateUserId(uuidv4());
       else if (type === ACTION_NEXT) {
+        // Wait for step submission to succeed
+        if (submitPromise && submitPromise !== undefined) {
+          try {
+            await submitPromise;
```

Comment:
we shouldn't do this. we very deliberately didn't await the promise here because we wanted navigation to be snappy and not depend on the submit returning. My proposal was that we just check the offline flag here instead